ledger/streaming/client/benches/bench.rs

```diff
@@ -0,0 +1,28 @@
+#![feature(test)]
+
+extern crate test;
+
+use futures::{executor::block_on, StreamExt};
+use mc_ledger_db::test_utils::MockLedger;
+use mc_ledger_streaming_api::{test_utils::get_custom_block_stream, BlockStream};
+use mc_ledger_streaming_client::block_validator::BlockValidator;
+use test::Bencher;
+
+/// Process 1000 typically sized blocks
+#[bench]
+fn process_1000_blocks(b: &mut Bencher) {
+    let logger = mc_common::logger::create_test_logger("benchmark_logger".to_string());
+    let input_stream_builder = get_custom_block_stream(1, 3, 1000, 2);
+    let ledger: Option<MockLedger> = None;
+    let output_stream_builder = BlockValidator::new(input_stream_builder, ledger, logger);
+
+    b.iter(|| {
+        let builder = output_stream_builder.clone();
+        block_on(async move {
+            let mut stream = builder.clone().get_block_stream(0).unwrap();
+            while let Some(_) = stream.next().await {
+                // Wait for benchmarks
```

Comment:
I'm not sure what this means

Reply:
In the bench test there's no need for any logic. The stream itself when its driven does the validation of the blocks and that's what the bench is testing. 

Before implementing the above changes - I'd like a discussion of approach first so that it's sanity checked. Once we have that decided, I'll bring the style up to production level.

The comment below this has the specific asks for commentary.